### PR TITLE
Eslint v9

### PR DESCRIPTION
**Pull request by tomi-timetac**
* Upgrade Eslint to v9
* refactor config to flat structure
* run fix on all files

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import tseslint from 'typescript-eslint';
+import eslintConfigPrettier from "eslint-config-prettier";
+// import eslintConfigImport from 'eslint-plugin-import';
+
+export default tseslint.config(
+  {
```
Comment (AlGvrl) [inline eslint.config.mjs:6]: I'm not sure how `tseslint.config` works, so i might be overlooking something, but I think the `extends` section of the old config is missing/not migrated.

https://eslint.org/docs/latest/use/configure/migration-guide#predefined-and-shareable-configs

**Review by AlGvrl — COMMENTED** [review]
Not sure if it's quite the same as before, left a couple of comments.

I can also take over this sometime soon once i sort things out in the internal repo if you don't have time to look into it.

**Comment by AlGvrl** [comment]
Pushed some changes, please review the tweaks @tomi-timetac

**Review by AlGvrl — APPROVED** [review]

**Comment by tomi-timetac** [comment]
LGTM

**Pull request merged**